packages/editor/src/EditorChildWindow.tsx

```diff
@@ -69,25 +67,28 @@ export function EasyblocksCanvas({
   }, []);
 
   useEffect(() => {
-    window.parent.editorWindowAPI.onUpdate = () => {
-      // Force re-render when child gets info from parent that data changed
-      forceRerender();
-    };
-  });
+    window.parent.postMessage(
+      {
+        type: "@easyblocks-editor/canvas-loaded",
+      },
+      "*"
+    );
+  }, []);
+
+  const canvasContext = useEasyblocksCanvasContext();
 
-  const shouldNotRender = !enabled || !meta || !compiled || !externalData;
+  const shouldNotRender = !enabled || !canvasContext;
```

Comment:
Since we check here for `canvasContext` there is no need to check if the context is defined in other places rendered within the canvas. Because of that check, we break the rules of hooks because we early return and thus render different amount of hooks between renders ex. RichText. We should introduce simply another hook `useEasyblocksCanvas` that uses `useEasyblocksCanvasContext` and throws if the context is undefined.
```
function useEasyblocksCanvas() {
  const canvasContext = useEasyblocksCanvasContext();
  if (!canvasContext) { throw new Error("...") }
  return canvasContext
}
```

Reply:
@pawel-commerce-studio please elaborate, not sure if your comment solves the issue here. Because the canvascontext is undefined initially, with postMessage it get's updated async. Maybe we should only render the children when a context is available?

Reply:
Yea, that's what I meant and we already do this by using `shouldNotRender` variable.